eng/pipelines/eng-common-sync.yml

```diff
@@ -30,16 +30,38 @@ stages:
         displayName: Sync ${{ parameters.DirectoryToSync }} Directory
 
         steps:
+        - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
```

Comment:
We should consider setting `Set-PsDebug -Trace 1` like https://github.com/Azure/azure-sdk-tools/blob/master/eng/pipelines/templates/steps/sync-repo-branch.yml#L25 for these larger inline pwsh scripts so that can more easily diagnose issues. This tracing will output every line of the script as it executes it.